resources-service/migrations/20230223150400-create-resource-string-attributes-table.js

```diff
@@ -0,0 +1,120 @@
+        "accountSid" text COLLATE pg_catalog."default" NOT NULL,
+        "key" text COLLATE pg_catalog."default" NOT NULL,
+        "value" text COLLATE pg_catalog."default" NOT NULL,
+        "language" text COLLATE pg_catalog."default" NOT NULL,
+        "info" JSONB,
+        "lastUpdated" timestamp with time zone,
+        "updateSequence"  bigint NOT NULL DEFAULT nextval('"Resources_updates_seq"'::regclass),
+        CONSTRAINT "ResourceStringAttributes_pkey" PRIMARY KEY ("resourceId", "accountSid", "key", "language", "value"),
+        CONSTRAINT "FK_ResourceStringAttributes_Resources" FOREIGN KEY ("resourceId", "accountSid")
+            REFERENCES resources."Resources" (id, "accountSid") MATCH SIMPLE
+            ON UPDATE CASCADE
+            ON DELETE CASCADE
+      )
+    `);
+    console.log('Table "ResourceStringAttributes" created');
+
+    await queryInterface.sequelize.query(`
+      ALTER TABLE IF EXISTS resources."ResourceStringAttributes"
+          OWNER to resources;
+    `);
+    console.log('Table "ResourceStringAttributes" now owned by resources');
+
+    await queryInterface.sequelize.query(`
+      CREATE FUNCTION resources."ResourcesLookupTables_updateSequence_trigger"()
+        RETURNS trigger
+        LANGUAGE 'plpgsql'
+        NOT LEAKPROOF
+      AS $BODY$
+      BEGIN
+        IF TG_WHEN <> 'BEFORE' THEN
+          RAISE EXCEPTION 'ResourcesLookupTables_updateSequence_trigger() may only run as an BEFORE trigger';
+        END IF;
+
+        IF (TG_LEVEL <> 'ROW' OR (TG_OP <> 'UPDATE')) THEN
+          RAISE EXCEPTION 'ResourcesLookupTables_updateSequence_trigger() added as trigger for unhandled case: %, %',TG_OP, TG_LEVEL;
+          RETURN NULL;
+        END IF;
+        
+        NEW."updateSequence" = nextval('"Resources_updates_seq"'::regclass);
+        RETURN NEW;
+      END
+      $BODY$;
+    `);
+    console.log('Function "ResourcesLookupTables_updateSequence_trigger" created.');
+
+    await queryInterface.sequelize.query(
+      `ALTER FUNCTION resources."ResourcesLookupTables_updateSequence_trigger"() OWNER TO resources`,
+    );
+    console.log('Function "ResourcesLookupTables_updateSequence_trigger" ownership altered.');
+
+    await queryInterface.sequelize.query(`
+      CREATE TRIGGER "ResourceStringAttributes_update_trigger"
+      BEFORE UPDATE
+      ON resources."ResourceStringAttributes"
+      FOR EACH ROW
+      WHEN (pg_trigger_depth() = 0)
+      EXECUTE FUNCTION resources."ResourcesLookupTables_updateSequence_trigger"();
+    `);
+    console.log('Trigger ResourceStringAttributes_update_trigger created');
+
+    await queryInterface.sequelize.query(`
+      CREATE TABLE IF NOT EXISTS resources."Globals"
+      (
+         -- This table is used to store global variables (i.e. not scoped to the account)
+         -- Having a boolean primary key with a CHECK constraint ensures that only one row is ever created
+        "singleRowId" bool NOT NULL PRIMARY KEY DEFAULT true,
```

Comment:
This is clever!

Question: is it the recommended way to implement this single-row constraint on relation DBs or is it just a clever workaround? I wish Postgres could have something native to express this, e.g. `CREATE SINGE ROW TABLE` or something.

Reply:
Follow up question: Other global values would be additional columns and the value itself would be part of this same row, correct?

Reply:
It's a bit bodgy but I couldn't find anything more idiomatic for Postgres. I suppose we could create a custom type that only has one possible value (like an enum with one entry or something) and use that for the PK, might be slightly neater but I'm not sure if custom types are supported on primary keys...

This is the simplest, seemingly most popular pattern I found but I could have missed something.

And yes, in the future I was thinking we could add more columns to the same table for things that aren't updated too often (like once a minute or so in this case). Anything that gets updated very frequently might be best off in it's own table to prevent possible locking contention